### PR TITLE
feat: @finnair/diff package with Diff and VersionInfo

### PR DESCRIPTION
**Pull request by ssaarela**
BREAKING CHANGE: More general and efficient PathMatcher API.

### CODE REVIEW

**Review by Paul53420 — COMMENTED** [review]
I am expecting @karlkeskyla to also review this code. On my behalf could not agree more and taking this package into use will reveal the potential

File: packages/diff/src/Diff.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  'bigint': true,
+  'symbol': true,
+};
+
+function isPrimitive(value: any) {
+  return value === null || value === undefined || !!primitiveTypes[typeof value]; 
+}
+
+export class Diff {
+
```
Comment (karlkeskyla) [inline packages/diff/src/Diff.ts:38]: Empty line could be removed (I got nothing else to complain about, so now I'm picking on empty lines 🤷‍♂️)

File: packages/diff/src/VersionInfo.spec.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import { describe, test, expect } from 'vitest';
+import { AnyIndex, Path, PathMatcher } from '@finnair/path';
+import { Change, Diff } from './Diff.js';
+import { VersionInfo, VersionInfoConfig } from './VersionInfo.js';
+
+describe('VersionInfo', () => {
+
```
Comment (karlkeskyla) [inline packages/diff/src/VersionInfo.spec.ts:7]: Could cut the empty line from here as well 🤔

**Review by karlkeskyla — APPROVED** [review]
Wicked!

Beautiful implementation, so all I could do is question the need of two empty lines! 🥇

**Review by karlkeskyla — APPROVED** [review]
Neat!

**Pull request merged**